client/verta/verta/operations/monitoring/profiler.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+import collections
+
+from verta.data_types import (
+    DiscreteHistogram,
+    FloatHistogram,
+)
+
+import numpy as np
+
+
+class Profiler(object):
+    def __init__(self, columns):
+        self.columns = columns
+
+    def profile(self, df):
+        return {column: histogram for (column, histogram) in self._profile_columns(df)}
+
+    # compatibility with model machinery
+    def predict(self, df):
+        return self.profile(df)
+
+    def _profile_columns(self, df):
+        return [self._profile_column(df, column) for column in self.columns]
+
+    def _profile_column(self, df, column):
+        raise NotImplementedError("")
+
+    def __name__(self):
+        return "Profiler"
```

Comment:
I wonder where/if this is being used? I'm concerned about messing with assigning `__name__` (conventionally, it's only defined by default on classes rather than instances)